pqarrow/arrow.go

```diff
@@ -17,20 +16,20 @@ import (
 	"github.com/polarsignals/frostdb/query/logicalplan"
 )
 
-// ParquetRowGroupToArrowRecord converts a parquet row group to an arrow record.
-func ParquetRowGroupToArrowRecord(
+// ParquetRowGroupToArrowSchema converts a parquet row group to an arrow schema.
+func ParquetRowGroupToArrowSchema(
 	ctx context.Context,
 	pool memory.Allocator,
 	rg parquet.RowGroup,
 	projections []logicalplan.ColumnMatcher,
 	filterExpr logicalplan.Expr,
 	distinctColumns []logicalplan.ColumnMatcher,
-) (arrow.Record, error) {
+) (*arrow.Schema, error) {
 	switch rg.(type) {
 	case *dynparquet.MergedRowGroup:
-		return rowBasedParquetRowGroupToArrowRecord(ctx, pool, rg)
+		return rowBasedParquetRowGroupToArrowSchema(ctx, rg)
```

Comment:
I don't see how for the schema row-based or contiguous makes a difference. The reason it makes a difference for the array conversion is that with contiguous ones, we know that the data to convert is all contained in a single contiguous page which is why we can exploit that nature for the conversion. The schema conversion should be identical to whether the underlying pages of column data is contiguous or not.